app/components/mandatarissen/mandataris-edit-prompt.js

```diff
@@ -24,8 +28,47 @@ export default class MandatenbeheerMandatarisEditPromptComponent extends Compone
     return this.editMode === CORRECT_MODE;
   }
 
-  get isBekrachtiging() {
-    return this.editMode === BEKRACHTIG_MODE;
+  get mandataris() {
+    return this.args.mandataris;
+  }
+
+  constructor() {
+    super(...arguments);
+
+    this.loadPublicationStatusOptions();
+  }
+
+  async getBekrachtigdStatus() {
+    return await queryRecord(this.store, 'mandataris-publication-status-code', {
+      'filter[:uri:]': MANDATARIS_BEKRACHTIGD_STATE,
+    });
+  }
+
+  isBekrachtigd(publicationStatus) {
+    return !publicationStatus || publicationStatus.isBekrachtigd;
+  }
+
+  async setBekrachtigdStatus() {
+    const bekrachtigdStatus = await this.getBekrachtigdStatus();
+
+    this.disabled = true;
+    this.selectedPublicationStatus = bekrachtigdStatus;
+  }
+
+  async loadPublicationStatusOptions() {
```

Comment:
This results in some weird behaviour when you update state. Then the old publicationStatusOptions are shown, might be better to fetch this in the component meant for the publicatieStatus, then you might be able to just invalidate the contents of the component instead of the whole route?